Docs: Explain how to run Mimir locally using docker-compose (#4079)
grafana/mimir: docs/internal/contributing/README.md

Grafana Mimir follows a standard GitHub pull request workflow. If you're unfamiliar with this workflow, read the very helpful [Understanding the GitHub flow](https://guides.github.com/introduction/flow/) guide from GitHub.

You are welcome to create draft PRs at any stage of readiness - this can be helpful to ask for assistance or to develop an idea. When you open a PR as a draft, it can be helpful to add a short description of what you're still working on and/or seeking assistance with. Before a piece of work is finished it should:

> - Be organised into one or more commits, each of which has a commit message that describes all changes made in that commit ('why' more than 'what' - we can read the diffs to see the code that changed).

Reviewer: ```suggestion
- Organize it into one or more commits, and include a commit message for each that describes all of the changes that you made in that commit. It is more helpful to explain _why_ more than _what_, which are available via `git diff`.
```

make CONTAINER_MOUNT_OPTIONS=delegated
```

> ### Running Grafana Mimir locally

Reviewer: ```suggestion
### Run Grafana Mimir locally
```

> An easy way to run Grafana Mimir locally during development is to use the docker-compose setup located at `development/<deployment-mode>/`, e.g.

Reviewer: ```suggestion
To easily run Grafana Mimir locally during development, use the docker-compose setup at `development/<deployment-mode>/`:
```

```bash

> development/mimir-read-write-mode/compose-up.sh -d

Reviewer 2: What is `-d` here? I've never used that flag.

Author: > 
--detach , -d |   | Detached mode: Run containers in the background
-- | -- | --

see [source](https://docs.docker.com/engine/reference/commandline/compose_up/),

I usually use it because most of the time I'm not interested in the logs of _all_ components, just the ones that have pending changes. Also, I was assuming this is what people use, because we've got the dedicated `compose-down.sh` script to stop containers, otherwise a `Ctrl+C` would do the job of that.

Reviewer 2: ah, thanks! you're right, I always use `ctrl+c` :) good to know about this option. That said, I'm not sure we need to include it in this doc, if default works fine.

Author: On the upside, if we do we also get to document the existence/purpose of the tear-down script. :) If you feel strongly about it, I'm happy to remove it.

Reviewer 2: I don't, feel free to merge as-is.

```

> This will build the local version of the code and start the Mimir components that are part of the chosen deployment mode as a set of containers. Use the `compose-down.sh` script to tear down the containers when they're no longer needed.

Reviewer: ```suggestion
Doing so builds the local version of the code and starts the Mimir components that are part of the chosen deployment mode as a set of containers. Use the `compose-down.sh` script to tear down the containers when you no longer need them.
```

## Documentation

> The Grafana Mimir documentation is compiled into a website published at [grafana.com](https://grafana.com/). Run `make docs` to build and serve the documentation locally. For more detail on style and organisation of the documentation, refer to the dedicated page "[How to write documenentation](how-to-write-documentation.md)".

Reviewer 2: Can we use Mimir-specific address, and spell it out fully like this: [https://grafana.com/docs/mimir/latest/](https://grafana.com/docs/mimir/latest/). It adds burden to us to keep it updated, but it also tells people where to find it.

Author: I could be missing something, but I don't think that file is part of the compiled documentation at all? Not sure if that is intended.

Reviewer 2: That link points to webpage generated from `docs/sources/mimir/_index.md` file in this repository. Perhaps I misunderstand what you mean?

Author: Ok, I got it now :) fixed in https://github.com/grafana/mimir/pull/4079/commits/f37e241606750f1250b676749f3853846b264d7b.

## Documentation

> The Grafana Mimir documentation is compiled into a website published at [https://grafana.com/docs/mimir/latest/](https://grafana.com/docs/mimir/latest/). Run `make docs` to build and serve the documentation locally.

Reviewer: ```suggestion
The Grafana Mimir documentation and the Helm chart for Mimir and GEM are compiled and published to [https://grafana.com/docs/mimir/latest/](https://grafana.com/docs/mimir/latest/) and [https://grafana.com/docs/helm-charts/mimir-distributed/latest/](https://grafana.com/docs/helm-charts/mimir-distributed/latest/). Run `make docs` to build and serve the documentation locally.
```
